src/smart-components/common/edit-approval-workflow.js

```diff
@@ -56,104 +32,46 @@ const EditApprovalWorkflow = ({
   };
 
-  const onSubmit = (formData) => {
+  const onSubmit = (toLink, toUnlink) => {
     close();
-    const unlinkArray = data
-      .filter(
-        ({ id }) =>
-          !formData['initial-workflows'].find((workflow) => id === workflow.id)
-      )
-      .map(({ id }) => id);
-    /**
-     * prevent uneccesary unlink and link API calls of the same workflow
-     */
-    const linkDiff = difference(formData['new-workflows'], unlinkArray);
-    const unLinkDiff = difference(unlinkArray, formData['new-workflows']);
-    const toLinkWorkflows = linkDiff.filter(
-      (id) => !data.find((item) => item.id === id)
-    );
-    const toUnlinkWorkflows = unLinkDiff.filter((id) =>
-      data.find((item) => item.id === id)
+    dispatch(
+      updateWorkflows(toLink, toUnlink, {
+        object_type: objectType,
+        app_name: APP_NAME[objectType],
+        object_id: query[querySelector]
+      })
     );
-    if (toUnlinkWorkflows.length > 0 || toLinkWorkflows.length > 0) {
-      dispatch(
-        updateWorkflows(toUnlinkWorkflows, toLinkWorkflows, {
-          object_type: objectType,
-          app_name: APP_NAME[objectType],
-          object_id: query[querySelector]
-        })
-      );
-    }
   };
 
-  const isLoadingFinal = isLoading || isFetching;
-
   return (
-    <Modal title={modalTitle} isOpen onClose={close} variant="small">
-      {isLoadingFinal && <WorkflowLoader />}
-      {!isLoadingFinal && (
-        <Stack hasGutter>
-          <StackItem>
-            <TextContent>
-              <Text>
-                {formatMessage(approvalMessages.setWorkflow, {
-                  // eslint-disable-next-line react/display-name
-                  strong: (chunks) => <strong>{chunks}</strong>,
-                  objectName: objectName(query[querySelector])
-                })}
-              </Text>
-            </TextContent>
-          </StackItem>
-          <StackItem>
-            <FormRenderer
-              subscription={{ values: true }}
-              initialValues={{
-                'initial-workflows': data
-              }}
-              onSubmit={onSubmit}
-              onCancel={close}
-              schema={editApprovalWorkflowSchema(loadWorkflowOptions)}
-              formContainer="modal"
-              templateProps={{
-                submitLabel: formatMessage(actionMessages.save)
-              }}
-            />
-          </StackItem>
-        </Stack>
-      )}
-    </Modal>
+    <TaggingModal
+      title={modalTitle}
+      onClose={close}
+      onSubmit={onSubmit}
+      getInitialTags={() =>
+        listWorkflowsForObject({
+          objectType,
+          appName: APP_NAME[objectType],
+          objectId: query[querySelector]
+        }).then(({ data }) => data)
+      }
+      loadTags={loadWorkflowOptions}
+      subTitle={formatMessage(approvalMessages.setWorkflow, {
+        // eslint-disable-next-line react/display-name
+        strong: (chunks) => <strong>{chunks}</strong>,
```

Comment:
In the share modal messages PR I exported this into a separate function. I think it could be moved to a shared file, so there would be no reason to write `// eslint-disable-next-line react/display-name` everywhere.

Reply:
It was the `b` tag right?

Reply:
Yeah, I think I called it 'bold'.. :smile: You can change it (and the element) to strong, it makes sense in our context.